src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -103,25 +105,27 @@ public HttpConnection(Uri url, ITransportFactory transportFactory, ILoggerFactor
 
         private Task StartAsyncCore()
         {
-            if (Interlocked.CompareExchange(ref _connectionState, ConnectionState.Connecting, ConnectionState.Initial)
-                != ConnectionState.Initial)
+            if (ChangeState(from: ConnectionState.Disconnected, to: ConnectionState.Connecting) != ConnectionState.Disconnected)
             {
                 return Task.FromException(
-                    new InvalidOperationException("Cannot start a connection that is not in the Initial state."));
+                    new InvalidOperationException("Cannot start a connection that is not in the Disconnected state."));
```

Comment:
`nameof(ConnectionState.Disconnected)`!

Reply:
Fixed. Initially it was not possible because state was tracked as int not an enum.